src/main/java/io/confluent/connect/hdfs/avro/AvroHiveUtil.java

```diff
@@ -91,9 +97,49 @@ private Table constructAvroTable(
       throw new HiveMetaStoreException("Cannot find input/output format:", e);
     }
     List<FieldSchema> columns = HiveSchemaConverter.convertSchema(schema);
+    removeFieldPartitionColumn(columns, partitioner.partitionFields());
     table.setFields(columns);
     table.setPartCols(partitioner.partitionFields());
-    table.getParameters().put(AVRO_SCHEMA_LITERAL, avroData.fromConnectSchema(schema).toString());
+    Schema filteredSchema = filterSchema(schema, partitioner.partitionFields());
+    table.getParameters().put(AVRO_SCHEMA_LITERAL,
+        avroData.fromConnectSchema(filteredSchema).toString());
     return table;
   }
+
+  /**
+   * Remove the column(s) that is later re-created by Hive when using the
+   * {@code partition.field.name} config.
+   *
+   * @param oldSchema the old schema to remove fields from
+   * @param partitionFields the fields used for partitioning
+   * @return the new schema without the fields used for partitioning
+   */
+  private Schema filterSchema(Schema oldSchema, List<FieldSchema> partitionFields) {
+    Set<String> partitions = partitionFields.stream()
+        .map(FieldSchema::getName).collect(Collectors.toSet());
+
+    SchemaBuilder newSchema = SchemaBuilder.struct();
+    for (Field f : oldSchema.fields()) {
+      if (!partitions.contains(f.name())) {
+        newSchema.field(f.name(), f.schema());
+      }
+    }
+    return newSchema;
+  }
+
+  /**
+   * Remove the column(s) that is later re-created by Hive when using the
+   * {@code partition.field.name} config.
+   *
+   * @param columns the hive columns from
+   *                {@link HiveSchemaConverter#convertSchema(Schema) convertSchema}.
+   * @param partitionFields the fields used for partitioning
+   */
+  private void removeFieldPartitionColumn(List<FieldSchema> columns,
+      List<FieldSchema> partitionFields) {
+    Set<String> partitions = partitionFields.stream()
+        .map(FieldSchema::getName).collect(Collectors.toSet());
+
+    columns.removeIf(column -> partitions.contains(column.getName()));
+  }
```

Comment:
we can move this method to `HiveUtil` instead of duplicating it across two classes

Reply:
I held off due to `HiveUtil` being `@Deprecated` and with the note 
```
// NOTE: DO NOT add or modify this class as it is maintained for compatibility
```

Reply:
Would be great indeed if we could add it there, wdyt @kkonstantine ?

Reply:
would have to add it here https://github.com/confluentinc/kafka-connect-storage-common/blob/master/hive/src/main/java/io/confluent/connect/storage/hive/HiveUtil.java then

Reply:
considering there the future work planned, im fine leaving as is